test/micro/org/openjdk/bench/java/math/BigDecimalDivide.java

```diff
@@ -0,0 +1,153 @@
+         * Each array entry is atmost 4096 bits
+         * in size
+         */
+        xlArray = new BigDecimal[TESTSIZE][2]; /*
+         * Each array entry is atmost 16384 bits
+         * in size
+         */
+
+        for (int i = 0; i < TESTSIZE; i++) {
+            xsArray[i][0] = new BigDecimal(new BigInteger(r.nextInt(64), r));
+            xsArray[i][1] = new BigDecimal(new BigInteger(r.nextInt(64), r));
+            sArray[i][0] = new BigDecimal(new BigInteger(r.nextInt(256), r));
+            sArray[i][1] = new BigDecimal(new BigInteger(r.nextInt(256), r));
+            mArray[i][0] = new BigDecimal(new BigInteger(r.nextInt(1024), r));
+            mArray[i][1] = new BigDecimal(new BigInteger(r.nextInt(1024), r));
+            lArray[i][0] = new BigDecimal(new BigInteger(r.nextInt(4096), r));
+            lArray[i][1] = new BigDecimal(new BigInteger(r.nextInt(4096), r));
+            xlArray[i][0] = new BigDecimal(new BigInteger(r.nextInt(16384), r));
+            xlArray[i][1] = new BigDecimal(new BigInteger(r.nextInt(16384), r));
+        }
+    }
+
+    /** Test BigDecimal.divide(BigDecimal) with numbers long at most 64 bits  */
+    @Benchmark
+    @OperationsPerInvocation(TESTSIZE)
+    public void testExactDivideXS(Blackhole bh) {
+        for (BigDecimal[] s : xsArray) {
+            try {
+                bh.consume(s[0].divide(s[1]));
+            } catch (ArithmeticException e) {
+            }
+        }
+    }
+
+    /** Test BigDecimal.divide(BigDecimal) with numbers long at most 256 bits */
+    @Benchmark
+    @OperationsPerInvocation(TESTSIZE)
+    public void testExactDivideS(Blackhole bh) {
+        for (BigDecimal[] s : sArray) {
+            try {
+                bh.consume(s[0].divide(s[1]));
+            } catch (ArithmeticException e) {
+            }
+        }
+    }
+
+    /** Test BigDecimal.divide(BigDecimal) with numbers long at most 1024 bits */
+    @Benchmark
+    @OperationsPerInvocation(TESTSIZE)
+    public void testExactDivideM(Blackhole bh) {
+        for (BigDecimal[] s : mArray) {
+            try {
+                bh.consume(s[0].divide(s[1]));
+            } catch (ArithmeticException e) {
+            }
+        }
+    }
+
+    /** Test BigDecimal.divide(BigDecimal) with numbers long at most 4096 bits */
+    @Benchmark
+    @OperationsPerInvocation(TESTSIZE)
+    public void testExactDivideL(Blackhole bh) {
+        for (BigDecimal[] s : lArray) {
+            try {
+                bh.consume(s[0].divide(s[1]));
+            } catch (ArithmeticException e) {
+            }
+        }
+    }
+
+    /** Test BigDecimal.divide(BigDecimal) with numbers long at most 16384 bits */
+    @Benchmark
+    @OperationsPerInvocation(TESTSIZE)
+    public void testExactDivideXL(Blackhole bh) {
+        for (BigDecimal[] s : xlArray) {
+            try {
+                bh.consume(s[0].divide(s[1]));
+            } catch (ArithmeticException e) {
+            }
+        }
+    }
+}
```

Comment:
It would be useful to publish the "before" and "after" results of running these benchmarks.